crates/turbopack-dev-server/src/source/resolve.rs

```diff
@@ -83,26 +72,55 @@ pub async fn resolve_source_request(
                                 );
                             }
                         }
-                        current_asset_path = new_asset_path;
-                        data = ContentSourceData::default();
-                    }
-                    ContentSourceContent::NotFound => {
-                        break Ok(ResolveSourceRequestResult::NotFound.cell());
+                        // do the rewrite
+                        match &rewrite.ty {
+                            RewriteType::Location { path_and_query } => {
+                                let new_uri = Uri::try_from(path_and_query)?;
+                                let new_asset_path =
+                                    urlencoding::decode(&new_uri.path()[1..])?.into_owned();
+                                request_overwrites.uri = new_uri;
+                                current_asset_path = new_asset_path;
+                                continue 'routes;
+                            }
+                            RewriteType::ContentSource {
+                                source,
+                                path_and_query,
+                            } => {
+                                let new_uri = Uri::try_from(path_and_query)?;
+                                let new_asset_path =
+                                    urlencoding::decode(&new_uri.path()[1..])?.into_owned();
+                                request_overwrites.uri = new_uri;
+                                current_asset_path = new_asset_path;
+                                route_tree =
+                                    source.get_routes().resolve_strongly_consistent().await?;
+                                continue 'routes;
+                            }
+                            RewriteType::Sources {
+                                sources: new_sources,
+                            } => {
+                                sources = *new_sources;
+                                continue 'sources;
+                            }
```

Comment:
And this creates similar inconsistencies. There could be more sources after the current source that would give a valid result, but we'll never reach those. I think we need to only `sources = new_sources.concat(remaining_sources)` here

Reply:
Same, the Rewrite response is counted as route match and no following source is applied.

e. g. for `/some/path` and `/some/[dynamic]` returned by `get_routes`, once `/some/path` returns a Rewrite, `/some/[dynamic]` will no longer be applied